src/pfe/portal/modules/Templates.js

```diff
@@ -72,67 +73,24 @@ module.exports = class Templates {
   }
 
   async getTemplatesFromRepos(repositoryList) {
-    let newProjectTemplates = [];
-
-    // apply reduce function to create a copy of the repository list index by url
-    const repos = repositoryList.reduce(reducer, {});
-
-    // query providers for more repositories
-    for (let provider of Object.values(this.providers)) {
-      try {
-        const extraRepos = await provider.getRepositories();
-        if (Array.isArray(extraRepos))
-          // apply reduce function here again, urls that we already have are ignored
-          extraRepos.reduce(reducer, repos);
-      }
-      catch (err) {
-        log.error(err.message);
-      }
-    }
-
-    await Promise.all(Object.values(repos).map(async function getTemplatesFromRepo(repository) {
-
-      let repositoryUrl;
-      try {
-        repositoryUrl = new URL(repository.url);
-      } catch (error) {
-        log.warn(error.message);
-        return; // ignore so that others can continue
-      }
-
-      let options = {
-        host: repositoryUrl.host,
-        path: repositoryUrl.pathname,
-        method: 'GET',
-      }
+    const providers = Object.values(this.providers);
+    const providedRepos = await getReposFromProviders(providers);
+    // Avoid processing duplicate repos
+    const extraRepos = providedRepos.filter(repo =>
+      !repositoryList.find(repo2 => repo2.url === repo.url)
```

Comment:
This is a little less efficient than how it was before, since for each provider repo we need to do a find on the master list.

Granted, those 2 list probably aren't going to be huge so it's probably fine but just want to mention this.